src/module/actor/data-model-classes/data-model-character-move.ts

```diff
@@ -0,0 +1,99 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import OseDataModelCharacterEncumbrance from "./data-model-character-encumbrance";
+
+export interface CharacterMove {
+  base: number;
+  encounter: number;
+  overland: number;
+}
+
+/**
+ * A class representing a character's move speeds.
+ */
+export default class OseDataModelCharacterMove implements CharacterMove {
+  static baseMoveRate = 120;
+  
+  #moveBase;
+  #autocalculate;
+  #encumbranceVariant;
+  #overEncumbranceLimit;
+
+  #heaviestArmor;
+  #overTreasureLimit;
+  
+  #halfEncumbered;
+  #quarterEncumbered;
+  #eighthEncumbered;
+  
+  /**
+   * 
+   * @param {OseDataModelCharacterEncumbrance} encumbrance An object representing the character's encumbrance values
+   * @param {boolean} shouldCalculateMovement Should the class autocalculate movement?
+   * @param {number} baseMoveRate The base move rate for the actor
+   */
+  constructor(
+    encumbrance: OseDataModelCharacterEncumbrance, 
```

Comment:
How do we feel about defaulting this to `new OseDataModelCharacterEncumbrance('disabled')`? We'd be able to use this class for non-characters, too.

Reply:
Seems reasonable!

I think it'd be interesting to think about something like a subclass like `OseDataModelCharacterDisableEncumbrance`. Certainly more work, but would give a good look at how module developers might implement functionality overrides

Reply:
Honestly, turning all of the encumbrance variants into subclasses feels like a decent idea -- if we're going to suggest folks take that approach in developing their own encumbrance schemes,we ought to dogfood it.